src/client/components/input/partial-date.js

```diff
@@ -25,7 +25,19 @@ const ymdRegex = /^\d{4}-\d{2}-\d{2}$/;
 const ymRegex = /^\d{4}-\d{2}$/;
 const yRegex = /^\d{4}$/;
 
+/**
+* This class is derived from the React Component base class
+* to render the partial date of any of the formats "YYYY", "YYYY-MM",
+* "YYYY-MM-DD" which allows the editor to be vague where information is not
+* known. It also checks validity of the value filled in the date field.
+*/
 class PartialDate extends React.Component {
+	/**
+	* Function to check validity of the date filled in the field.
```

Comment:
"... validity of a partial date" - this is a static function, so it's not really linked to the field at all.